pkg/security/podsecuritypolicy/seccomp/strategy_test.go

```diff
@@ -161,57 +167,90 @@ func TestValidatePod(t *testing.T) {
 	tests := map[string]struct {
 		pspAnnotations map[string]string
 		podAnnotations map[string]string
+		seccompProfile *api.SeccompProfile
 		expectedError  string
 	}{
 		"no pod annotations, required profiles": {
 			pspAnnotations: allowSpecific,
 			podAnnotations: nil,
+			seccompProfile: nil,
 			expectedError:  "Forbidden:  is not an allowed seccomp profile. Valid values are foo",
 		},
 		"no pod annotations, no required profiles": {
 			pspAnnotations: withoutSeccomp,
 			podAnnotations: nil,
+			seccompProfile: nil,
 			expectedError:  "",
 		},
 		"valid pod annotations, required profiles": {
 			pspAnnotations: allowSpecific,
 			podAnnotations: map[string]string{
-				api.SeccompPodAnnotationKey: "foo",
+				api.SeccompPodAnnotationKey: foo,
 			},
-			expectedError: "",
+			seccompProfile: nil,
+			expectedError:  "",
 		},
 		"invalid pod annotations, required profiles": {
 			pspAnnotations: allowSpecific,
 			podAnnotations: map[string]string{
-				api.SeccompPodAnnotationKey: "bar",
+				api.SeccompPodAnnotationKey: bar,
 			},
-			expectedError: "Forbidden: bar is not an allowed seccomp profile. Valid values are foo",
+			seccompProfile: nil,
+			expectedError:  "Forbidden: bar is not an allowed seccomp profile. Valid values are foo",
 		},
 		"pod annotations, no required profiles": {
 			pspAnnotations: withoutSeccomp,
 			podAnnotations: map[string]string{
-				api.SeccompPodAnnotationKey: "foo",
+				api.SeccompPodAnnotationKey: foo,
 			},
-			expectedError: "Forbidden: seccomp may not be set",
+			seccompProfile: nil,
+			expectedError:  "Forbidden: seccomp may not be set",
 		},
 		"pod annotations, allow any": {
 			pspAnnotations: allowAnyNoDefault,
 			podAnnotations: map[string]string{
-				api.SeccompPodAnnotationKey: "foo",
+				api.SeccompPodAnnotationKey: foo,
 			},
-			expectedError: "",
+			seccompProfile: nil,
+			expectedError:  "",
 		},
 		"no pod annotations, allow any": {
 			pspAnnotations: allowAnyNoDefault,
 			podAnnotations: nil,
+			seccompProfile: nil,
 			expectedError:  "",
 		},
+		"valid pod annotations and field, required profiles": {
+			pspAnnotations: allowSpecific,
+			podAnnotations: map[string]string{
+				api.SeccompPodAnnotationKey: foo,
+			},
+			seccompProfile: &api.SeccompProfile{
+				Type:             api.SeccompProfileTypeLocalhost,
+				LocalhostProfile: &foo,
+			},
+			expectedError: "",
+		},
+		"valid pod field and no annotation, required profiles": {
+			pspAnnotations: allowSpecific,
+			podAnnotations: nil,
+			seccompProfile: &api.SeccompProfile{
+				Type:             api.SeccompProfileTypeLocalhost,
+				LocalhostProfile: &foo,
+			},
+			expectedError: "Forbidden: localhost/foo is not an allowed seccomp profile. Valid values are foo",
+		},
 	}
```

Comment:
add another testcase where just the field is set and is allowed (`pspAnnotations: allowSpecificLocalhost,`)

Reply:
Done :checkered_flag: